var/spack/repos/builtin/packages/python/package.py

```diff
@@ -272,7 +272,7 @@ def determine_version(cls, exe):
         # Output looks like:
         #   Python 3.7.7
         output = Executable(exe)('-V', output=str, error=str)
-        match = re.search(r'Python\s+(\S+)', output)
+        match = re.search(r'Python\s+([0-9a-zA-Z.\-]+)', output)
```

Comment:
No need to escape the `-` if it comes last. Also, you can add `_` if you want to match `spack.versions.VALID_VERSION`.

Reply:
Ah, it clamps to the begin and end, that's inconvenient.

Reply:
Reworked.